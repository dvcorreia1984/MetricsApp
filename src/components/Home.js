// Home.js

import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { BsArrowRightCircle } from 'react-icons/bs';
import SouthernAfricaImage from '../assets/country.svg';
import NorthernAfricaImage from '../assets/country2.svg';
import AfricaImage from '../assets/africa.svg';
import { fetchSubregionData } from '../redux/country/countrySlice';

const Home = () => {
  const dispatch = useDispatch();
  const [southernAfricaData, northernAfricaData] = useSelector((state) => [
    state.countries.southernAfricaData,
    state.countries.northernAfricaData,
  ]);

  useEffect(() => {
    dispatch(fetchSubregionData('Southern Africa'));
    dispatch(fetchSubregionData('Northern Africa'));
  }, [dispatch]);

  // Function to calculate the total population of a subregion
  const calculatePopulation = (data) => {
    if (data) {
      return data.population;
    }
    return 'Loading...';
  };

  return (
    <div>
      <div className="flex bg-[#5487e3] justify-evenly">
        <div className="w-1/4 text-blue-950 p-5">
          <img src={AfricaImage} alt="africa" />
        </div>
        <div className="flex flex-col justify-center items-start text-white">
          <h1 className="font-bold text-4xl">AFRICA</h1>
          <h2>5.6780 views</h2>
        </div>
      </div>
      <div className="bg-[#35548b] pl-5 text-white font-bold">STATS BY COUNTRY</div>
      <div className="flex flex-wrap justify-center bg-[#3f66ab]">
        {/* Container div with flex styles */}
        <div className="flex justify-between w-full max-w-screen-xl mx-auto p-4">
          <Link to="/southern-africa" className="w-full sm:w-1/2 m-2 relative">
            <BsArrowRightCircle className="text-white text-2xl absolute top-0 right-0" />
            <div className="w-full aspect-square text-white font-bold p-5 flex flex-col items-center">
              <img
                src={SouthernAfricaImage}
                alt="Southern Africa"
                className="w-2/3 h-2/3 mx-auto object-cover"
              />
              <div className="w-full flex flex-col">
                <span className="flex justify-end">SOUTHERN</span>
                <span className="flex justify-end">AFRICA</span>
                <span className="flex justify-end">
                  {calculatePopulation(southernAfricaData)}
                </span>
              </div>
            </div>
          </Link>
          <Link to="/northern-africa" className="w-full sm:w-1/2 m-2 relative">
            <BsArrowRightCircle className="text-white text-2xl absolute top-0 right-0" />
            <div className="w-full aspect-square text-white font-bold p-5 flex flex-col items-center">
              <img
                src={NorthernAfricaImage}
                alt="Northern Africa"
                className="w-2/3 h-2/3 mx-auto object-cover"
              />
              <div className="w-full flex flex-col">
                <span className="flex justify-end">NORTHERN</span>
                <span className="flex justify-end">AFRICA</span>
                <span className="flex justify-end">
                  {calculatePopulation(northernAfricaData)}
                </span>
              </div>
            </div>
          </Link>
        </div>
      </div>
    </div>
  );
};

export default Home;
