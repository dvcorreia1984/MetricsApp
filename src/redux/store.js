// store.js
import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './country/countrySlice';

export const store = configureStore({
  reducer: {
    countries: countryReducer,
  },
});
