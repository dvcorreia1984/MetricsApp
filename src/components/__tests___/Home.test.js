import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Home from '../Home'; // Adjust the import path based on your

describe('Home Component Snapshot', () => {
  it('should match the snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
