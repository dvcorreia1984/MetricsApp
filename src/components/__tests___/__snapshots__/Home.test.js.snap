// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Snapshot should match the snapshot 1`] = `
<div>
  <div
    className="flex bg-[#5487e3] justify-center gap-10"
  >
    <div
      className="w-1/4 text-blue-950 py-3"
    >
      <img
        alt="africa"
        src="africa.svg"
      />
    </div>
    <div
      className="flex flex-col justify-center items-start text-white"
    >
      <h1
        className="font-bold"
      >
         AFRICA 
      </h1>
      <h3
        className="mt-[-5px] text-[10px]"
      >
        Population:  60,546,598
      </h3>
    </div>
  </div>
  <div
    className="bg-[#35548b] pl-5 text-white text-sm"
  >
    STATS BY COUNTRY
  </div>
  <div
    className="flex justify-center bg-[#3f66ab]"
    id="container"
  >
    <div
      className="flex flex-wrap max-w-screen-xl mx-auto"
    />
  </div>
</div>
`;
